src/connector/src/sink/iceberg/mod.rs

```diff
@@ -346,6 +347,55 @@ impl IcebergConfig {
     }
 }
 
+pub async fn create_catalog(config: &IcebergConfig) -> Result<CatalogRef> {
+    match config.catalog_type() {
+        "storage" | "rest" => {
+            let iceberg_configs = config.build_iceberg_configs()?;
+            let catalog = load_catalog(&iceberg_configs)
+                .await
+                .map_err(|e| SinkError::Iceberg(anyhow!(e)))?;
+            Ok(catalog)
+        }
+        catalog_type if catalog_type == "hive" || catalog_type == "sql" || catalog_type == "glue" || catalog_type == "dynamodb" => {
+            // Create java catalog
+            let (base_catalog_config, java_catalog_props) = config.build_jni_catalog_configs()?;
+            let catalog_impl = match catalog_type {
+                "hive" => "org.apache.iceberg.hive.HiveCatalog",
+                "sql" => "org.apache.iceberg.jdbc.JdbcCatalog",
+                "glue" => "org.apache.iceberg.aws.glue.GlueCatalog",
+                "dynamodb" => "org.apache.iceberg.aws.dynamodb.DynamoDbCatalog",
+                _ => unreachable!(),
+            };
+
+            jni_catalog::JniCatalog::build(base_catalog_config, "risingwave", catalog_impl, java_catalog_props)
+        }
+        _ => {
+            Err(SinkError::Iceberg(anyhow!(
+                "Unsupported catalog type: {}, only support `storage`, `rest`, `hive`, `sql`, `glue`, `dynamodb`",
+                config.catalog_type()
+            )))
+        }
+    }
+}
+
+pub async fn create_table(config: &IcebergConfig) -> Result<Table> {
```

Comment:
Because we need to reuse this function in frontend to extract the partition spec of a table. 
https://github.com/risingwavelabs/risingwave/blob/0cc5ba92ce82b352d7195d1056c4a1dadb5edb8f/src/frontend/src/handler/create_sink.rs#L324
We can make the `create_table` pub only.